src/backend/sgx/thread.rs

```diff
@@ -57,8 +71,71 @@ impl super::super::Keep for super::Keep {
     }
 }
 
+fn sgx_enarxcall<'a>(enarxcall: &'a mut Payload, data: &'a mut [u8]) -> Option<Item<'a>> {
```

Comment:
I believe the signature of this function is an artifact of previous iteration using `filter_map`?
Since we're just looping over the items now and passing stuff individually to the host executor, why not return a `Result` here and propagate the errors to the caller rather than panicking inside? I would expect that these errors would (eventually) be rich and contain, e.g. a trace (I don't know if we already have a primitive for that)

Reply:
yes, changed